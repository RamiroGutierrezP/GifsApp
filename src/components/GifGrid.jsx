import { GifItem } from "./GifItem";
import PropTypes from 'prop-types';
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
    
    const {images, isLoading} = useFetchGifs (category);

    return (
        <>
            <h3>{ category }</h3>

            {
                isLoading && (<h2>Cargando...</h2>)
            }

            <div className="card-grid">
                {
                    images.map ( (image) => (
                        <GifItem 
                            key={ image.id }
                            //Uso el spread para que el GifItem reciba cada propiedad de la image como una property 
                            {...image}
                        />
                    ))
                }
            </div>
        </>
  )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}