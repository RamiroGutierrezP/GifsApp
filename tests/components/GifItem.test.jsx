import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('GifItem tests', () => { 
    
    const title ='Nappa';
    const url ='https://media.giphy.com/media/3o7btNa0RUYa5E7iiQ/giphy.gif';
    
    test ('should match with snapshot', () => {

        const { container } = render(<GifItem title = { title } url={ url }/>);
        
        expect(container).toMatchSnapshot();
    })

    test ('should show the image', () => {

        render(<GifItem title = { title } url={ url }/>);

        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

 })