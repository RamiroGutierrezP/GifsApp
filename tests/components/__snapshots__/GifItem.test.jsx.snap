// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem tests should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Nappa"
      src="https://media.giphy.com/media/3o7btNa0RUYa5E7iiQ/giphy.gif"
    />
  </div>
</div>
`;
